src/svc.rs

```diff
@@ -1,12 +1,63 @@
+extern crate linkerd2_router as rt;
 pub extern crate linkerd2_stack as stack;
 pub extern crate linkerd2_timeout;
 
 pub use self::linkerd2_timeout::stack as timeout;
 pub use self::stack::{layer, shared, Layer, LayerExt};
-pub use tower::builder::ServiceBuilder;
 pub use tower::util::{Either, Oneshot};
 pub use tower::{service_fn as mk, MakeConnection, MakeService, Service, ServiceExt};
 
-pub fn builder() -> ServiceBuilder<tower::layer::util::Identity> {
-    ServiceBuilder::new()
+use std::time::Duration;
+use tower::builder::ServiceBuilder;
+use tower::layer::util::{Identity, Stack};
+use tower::limit::concurrency::ConcurrencyLimitLayer;
+use tower::load_shed::LoadShedLayer;
+use tower::timeout::TimeoutLayer;
+
+use proxy::{buffer, pending};
+
+#[derive(Clone, Debug)]
+pub struct Builder<L>(ServiceBuilder<L>);
+
+pub fn builder() -> Builder<Identity> {
+    Builder(ServiceBuilder::new())
+}
+
+impl<L> Builder<L> {
```

Comment:
I think I like this? Seems like it will keep noise down in `app/main.rs` when we're building stacks, but it also introduces another place where we can define layer logic.

Reply:
Yeah, I see it as a place for capturing repeated patterns we use in the main. It should be used sparingly, though we certainly may want to split out more common stuff into this.